controllers/datadogmonitor/controller.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	// Validate the DatadogMonitor spec
+	if err = datadoghqv1alpha1.IsValidDatadogMonitor(&instance.Spec); err != nil {
+		logger.Error(err, "Invalid DatadogMonitor spec")
+		return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+
+	}
+
+	instanceSpecHash, err := comparison.GenerateMD5ForSpec(&instance.Spec)
+	if err != nil {
+		logger.Error(err, "Error generating hash", "err", err)
+		return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+
+	}
+	statusSpecHash := instance.Status.CurrentHash
+
+	// Create or update monitor, or check monitor state. Fall through this block (without returning)
+	// if the result should be requeued with the default period
+	if instance.Status.ID == 0 {
+		// If the monitor ID is 0, then it doesn't exist yet in Datadog. Create the monitor (only metric alerts)
+		if string(instance.Spec.Type) != string(datadogapiclientv1.MONITORTYPE_METRIC_ALERT) {
+			err = errors.New("monitor type not supported")
+			logger.Error(err, "For the alpha version, only metric alert type monitors are supported")
+			return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+		}
+		// Make sure required tags are present
+		if result, err = r.checkRequiredTags(logger, instance); err != nil || result.Requeue {
+			return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+		}
+
+		err = r.create(logger, instance, newStatus, now)
+		if err != nil {
+			logger.Error(err, "Error creating monitor")
+		}
+		newStatus.CurrentHash = instanceSpecHash
+	} else {
+		// Check if instance needs to be updated
+		if instanceSpecHash != statusSpecHash {
+			// Make sure required tags are present
+			if result, err = r.checkRequiredTags(logger, instance); err != nil || result.Requeue {
+				return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+			}
+			// Update action
+			err = r.update(logger, instance, newStatus, now)
+			if err != nil {
+				logger.Error(err, "Error updating monitor")
+			}
+			newStatus.CurrentHash = instanceSpecHash
+		} else {
+			// Spec has not changed, just check if monitor state has changed (alert, warn, OK, etc.)
+			err = r.get(logger, instance, newStatus)
+			if err != nil {
+				logger.Error(err, "Error getting monitor")
+			}
+		}
+	}
+
+	// Requeue
+	if !result.Requeue && result.RequeueAfter == 0 {
+		result.RequeueAfter = defaultRequeuePeriod
+	}
+
+	// Update the status via the Kubernetes API
+	return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+}
+
+func (r *Reconciler) create(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+	logger.Info(fmt.Sprint(datadogMonitor.Spec.Query))
+
+	logger.Info(fmt.Sprint(datadogMonitor.Status.ID))
+	// Validate monitor in Datadog
+	err := validateMonitor(r.datadogAuth, r.datadogClient, datadogMonitor)
+	if err != nil {
+		return err
+	}
```

Comment:
to scope the error to the if condition
```suggestion
	if err := validateMonitor(r.datadogAuth, r.datadogClient, datadogMonitor); err != nil {
		return err
	}
```